solutionbox/code_free_ml/analyze_data.py

```diff
@@ -78,7 +84,67 @@ def parse_arguments(argv):
     ValueError: for bad parameters
   """
   parser = argparse.ArgumentParser(
-      description='Runs analysis on structured data.')
+      formatter_class=argparse.RawDescriptionHelpFormatter,
+      description=textwrap.dedent("""\
+          Runs analysis on structured data and produces auxiliary files for
+          training. The output files can also be used by the Transform step
+          to materialize TF.Examples files, which for some problems can speed up
+          training.
+
+          Description of input files
+          --------------------------
+
+          1) If using csv files, the --csv-schema-file must be the file path to
+             a schema file. The format of this file must be a valid BigQuery
+             schema file, which is a JSON file containing a list of dicts.
+             Consider the example schema file below:
+
+             [
+                {"name": "column_name_1", "type": "integer"},
+                {"name": "column_name_2", "type": "float"},
+                {"name": "column_name_3", "type": "string"},
+                {"name": "column_name_4", "type": "string"},
+             ]
+
+             Note that the column names in the csv file much match the order
+             in the schema list. Also, we only support three BigQuery types (
+             integer, float, and string).
+
+             If instead of csv files, --bigquery-table is used, the schema file
+             does not have to be pass in as this program will extract it from
```

Comment:
"the schema file does not have to be pass in" --> "the schema file is not needed".